### ISSUE TITLE
Per-update unique Git commit message?

### ISSUE THREAD

**Issue by TonyGravagno** [zokugun/vscode-sync-settings#67]
I'm sorry if I've missed this in the documentation. Does this extension provide a way to open a UI prompt to change the commit message? Or does that need to be done in the git console from the shell?

If that isn't an option, is it a valid request to allow for prompting for the commit message, and only on default use the `{{profile}}` default?

**Comment by daiyam** [comment]
You can customize the commit messages: https://github.com/zokugun/vscode-sync-settings/blob/master/docs/commit-messages.md

Is that enough?

**Comment by TonyGravagno** [comment]
No, thanks for asking. With any project, a commit message tells us what the commit is about, not just when it happened. For a change to settings we might want to be explicit in the commit message about what changed.

The specific scenario that led to this request is that I am changing details in my theme. In the past I know when I have done this I later ask myself "why did I do that?" or even "what does that setting actually do?". If we make a change like that and then commit with a custom message, the reason is recorded.

So I'm looking for a way to prompt for a custom message, and if there is none, default to "syncSettings.gitUpdateMessage".

**Comment by daiyam** [comment]
> The specific scenario that led to this request is that I am changing details in my theme. In the past I know when I have done this I later ask myself "why did I do that?" or even "what does that setting actually do?". If we make a change like that and then commit with a custom message, the reason is recorded.

Ok, it what I was expecting your request was about. I think it's a normal request.

For the time being, you could use `local-git` (but you have issue with hooks) or use `file` and manually push/pull...

**Comment by TonyGravagno** [comment]
Got it. TY
